updatesnap/unittests.py

```diff
@@ -423,6 +428,53 @@ def test_invalid_version_variation_and_beta_release(self):
                 part['part_name'], part['version'], part['entry_format'], False)
             assert version is None
 
+    def test_snap_version_automation(self):
+        """ tests if snap version automation working correctly"""
+        data = self._base_load_test_file("test_snap_version_automation.yaml")
+        yaml_obj = ManageYAML(data)
+        snap, _, _, _ = self._load_test_file("test_snap_version_automation.yaml",
+                                             None)
+        args = Namespace(
+            version_schema=r'^gutenprint-(\d+_\d+_\d+)',
+        )
+        logging.basicConfig(level=logging.ERROR)
+        test = is_version_update(snap, yaml_obj, args)
+        os.remove('version_file')
+        assert test is not None
+
+    def test_no_version_in_metadata(self):
+        """ tests if snap version automation fails
+             if version is not present in metadata"""
+        data = self._base_load_test_file("test_no_version_in_metadata.yaml")
+        yaml_obj = ManageYAML(data)
+        snap, _, _, _ = self._load_test_file("test_no_version_in_metadata.yaml",
+                                             None)
+        args = Namespace(
+            version_schema=r'^debian/(\d+\.\d+\.\d+)',
+        )
+        logging.basicConfig(level=logging.ERROR)
+        test = is_version_update(snap, yaml_obj, args)
+        assert not test
+
+    def test_correct_snap_version(self):
+        """ tests if snap version number is correct or not"""
+        contents = self._base_load_test_file("test_snap_version_automation.yaml")
+        manager_yaml = ManageYAML(contents)
+        snap, _, _, _ = self._load_test_file("test_snap_version_automation.yaml",
+                                             None)
+        args = Namespace(
+            version_schema=r'^gutenprint-(\d+_\d+_\d+)',
+        )
+
+        def mock_process_version_data(_git_repo_url, _snap_name, _version_schema):
+            return "5.3.4-1"
+
+        temp = copy.copy(snap_version_module.process_snap_version_data)
```

Comment:
I think that you don't need to use copy.copy, just assign the function to "temp"...